src/Maestro/DependencyUpdateErrorProcessor/Program.cs

```diff
@@ -53,16 +52,36 @@ private static void Main()
                                     IConfigurationSection section = config.GetSection("GitHub");
                                     section.Bind(options);
                                 });
-
                             services.Configure<DependencyUpdateErrorProcessorOptions>(
                                 (options, provider) =>
-                            {
-                                var config = provider.GetRequiredService<IConfiguration>();
-                                options.IsEnabled = bool.Parse(config["EnableDependencyUpdateErrorProcessor"]);
-                                options.GithubUrl = config["GithubUrl"];
-                                options.FyiHandle = config["FyiHandle"];
-                            });
+                                {
+                                    var config = provider.GetRequiredService<IConfiguration>();
+                                    options.IsEnabled = bool.Parse(config["EnableDependencyUpdateErrorProcessor"]);
+                                    options.GithubUrl = config["GithubUrl"];
+                                    options.FyiHandle = config["FyiHandle"];
+                                });
                         });
+                    host.ConfigureContainer(builder => builder.Register((c, p) =>
```

Comment:
This won't with the validation I just checked in... this is going to fail your PR.  Please avoid the Autofac stuff unless you have no choice.

We definition shouldn't be starting a bunch of threads and waiting for them here.  This is bad.

Reply:
Actually, you're going to have a pretty massive merge conflict in this file (since I moved all the code in it), probably want to merge that in and see what's up.

Reply:
The only other way to do this is with a factory/provider interface. Like IGitHubClientFactory.

We use the autofac stuff so we can parameterize the resolution because we need a separate GitHubClient per repo. This can also be done with an interface and implementing class, but I didn't want to make a whole new class for this.

Reply:
Well, it's still not going to pass validation, so we don't have much choice now.  The tests won't be able to resolve the func.  We also _already_ have a class for this, so let's just use it.

Reply:
I'd rather the code be in a nice, encapsulated, reusable chunk as well.  This isn't the only service that needs to interact with github, let's not make everyone copy paste a bunch of sync-over-async code into their startup methods.

Reply:
Ooo, but it will. Because this thing isn't in the collection, it won't get validated...  that's... probably not good (and I'm going to fix that right quick, so don't depend on it).

Reply:
Weird triple actually, there aren't any tests for this assembly, so I didn't add my test to them, so they aren't testing that their DI works at all.

We should add that test.

Reply:
> We also _already_ have a class for this

where is this class we already have? I don't remember it existing.

Reply:
It's in a stupid place, src\DotNet.Status.Web\Controllers\GitHubClientFactory.cs.  It probably should get moved out to the Shared\*.GitHub.Authentication library.  I'm not sure what I didn't just put it there to start with, honestly.

Reply:
> Actually, you're going to have a pretty massive merge conflict in this file (since I moved all the code in it), probably want to merge that in and see what's up.

Hey @ChadNedzlek , I looked your commit https://github.com/dotnet/arcade-services/commit/c6a049cf6522bfb16a55a0c27b0e82ff046a7ef9, there is no change to my file in that.

Reply:
Indeed, because there are no tests for it, so I didn't have to refactor it to make those tests...

I'm in the middle of removing all references to Autofac in our code, there will probably be a PR later today.  There is a bunch of missed validation because of it, and I'm trying to button that up.

Reply:
I m not blocked now, but I will check in my changes after I get your changes later today. So that there wont be any conflicts.

Reply:
I checked in my changes with part of Chad's changes. I'm going to resolve this comment.